### PR TITLE
Use core connections API in processing database algorithms instead of Python versions

### PR DESCRIPTION
**Pull request by nyalldawson**
Ports the Postgres and Spatialite algorithms to the core database connection API instead of Python versions of these.

The python classes are basically 1:1 duplicates of the core classes, without unit tests and without a common API.

Part of the ongoing Processing database handling cleanup work, but also adds nice API for creation of spatial indexes which can be later used as part of the db manager cleanup/porting efforts.

Refs NRCan Contract#3000707093

### CODE REVIEW

**Review by alexbruy — APPROVED** [review]

File: python/plugins/processing/algs/gdal/ogr2ogrtopostgislist.py [hunk 1]
```diff
@@ -244,7 +255,7 @@ def getConsoleCommands(self, parameters, context, feedback, executing=True):
         arguments.append('-f')
         arguments.append('PostgreSQL')
         arguments.append('PG:"')
-        for token in uri.connectionInfo(executing).split(' '):
+        for token in QgsDataSourceUri(uri).connectionInfo(executing).split(' '):
```
Comment (elpaso) [inline python/plugins/processing/algs/gdal/ogr2ogrtopostgislist.py]: Shouldn't use `decodeUri` here?

**Review by elpaso — APPROVED** [review]
Maybe you can use `decodeUri` in some places instead of splitting on ` ` but I didn't check if it's really appropriate in this case.

File: python/plugins/processing/algs/gdal/ogr2ogrtopostgislist.py (hunk 1 above)
Comment (nyalldawson) [reply python/plugins/processing/algs/gdal/ogr2ogrtopostgislist.py]: Probably, but I'm reluctant to change anything I don't have to here :smiley:

**Comment by nirvn** [comment]
@nyalldawson , nice work.

**Pull request merged**